platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/legacy/AddressBookRoster.java

```diff
@@ -0,0 +1,149 @@
+import com.swirlds.common.platform.NodeId;
+import com.swirlds.platform.crypto.KeysAndCerts;
+import com.swirlds.platform.roster.Roster;
+import com.swirlds.platform.roster.RosterEntry;
+import com.swirlds.platform.system.address.Address;
+import com.swirlds.platform.system.address.AddressBook;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A {@link Roster} implementation that uses an {@link AddressBook} as its backing data structure.
+ */
+public class AddressBookRoster implements Roster {
+    private static final long CLASS_ID = 0x7104f97d4e298619L;
+
+    private static final class ClassVersion {
+        public static final int ORIGINAL = 1;
+    }
+
+    private final Map<NodeId, RosterEntry> entries = new HashMap<>();
+
+    /**
+     * Constructs a new {@link AddressBookRoster} from the given {@link AddressBook} and {@link KeysAndCerts} map.
+     *
+     * @param addressBook     the address book
+     * @param keysAndCertsMap the keys and certs map
+     */
+    public AddressBookRoster(
+            @NonNull final AddressBook addressBook, @NonNull final Map<NodeId, KeysAndCerts> keysAndCertsMap) {
+        Objects.requireNonNull(addressBook);
+        Objects.requireNonNull(keysAndCertsMap);
+
+        for (final Address address : addressBook) {
+            entries.put(address.getNodeId(), new AddressRosterEntry(address, keysAndCertsMap.get(address.getNodeId())));
+        }
+    }
+
+    /**
+     * Empty constructor for deserialization.
+     */
+    public AddressBookRoster() {}
+
+    @Override
+    public long getClassId() {
+        return CLASS_ID;
+    }
+
+    @Override
+    public int getVersion() {
+        return ClassVersion.ORIGINAL;
+    }
+
+    @Override
+    public void serialize(@NonNull SerializableDataOutputStream out) throws IOException {
+        out.writeInt(entries.size());
+        for (final RosterEntry entry : entries.values()) {
+            out.writeSerializable(entry, true);
+        }
+    }
+
+    @Override
+    public void deserialize(@NonNull SerializableDataInputStream in, int version) throws IOException {
+        final int size = in.readInt();
+        for (int i = 0; i < size; i++) {
+            final RosterEntry entry = in.readSerializable();
+            entries.put(entry.getNodeId(), entry);
+        }
+    }
+
+    @Override
+    @NonNull
+    public Collection<NodeId> getNodeIds() {
+        return entries.keySet();
+    }
+
+    @Override
+    @NonNull
+    public RosterEntry getEntry(@NonNull final NodeId nodeId) {
+        Objects.requireNonNull(nodeId);
+        return entries.get(nodeId);
+    }
+
+    @Override
+    @NonNull
+    public Iterator<RosterEntry> iterator() {
+        return new Iterator<RosterEntry>() {
```

Comment:
optional: remove explicit generic and replace with `<>`

Reply:
opted in.